M2/Macaulay2/packages/Cremona.m2

```diff
@@ -54,7 +54,8 @@ export{
    "abstractRationalMap",
    "specialCubicTransformation",
    "isMorphism",
-   "exceptionalLocus"
+   "exceptionalLocus",
+   "inverse'"
```

Comment:
The introduction of  `inverse'` will cause confusion with `inverse`.  Just add a new method to `inverse` instead.

Reply:
I already added the method inverse(RationalMap). Maybe I could replace inverse'(RationalMap) with something like inverse(RationalMap,Nothing)... or rename inverse' into inverse2 ?

Reply:
Let me help you find a better way.  What's the difference between `inverse(RationalMap)` and `inverse'(RationalMap)`?

Reply:
Both compute the inverse in the same way, but neither verifies whether the map is really birational (as this requires non-trivial computations). What change is just the test done at the end to verify that the output is correct and a fortiori that the input map was birational. inverse' performs a probabilistic but very fast test.

Reply:
And your method for `inverse` performs no test at all?

Reply:
`inverse` performs a non-probabilistic test but sometimes very hard.

Reply:
In any case, this sort of thing is exactly why we have optional named arguments to some method functions.  Writing something like `inverse(f,Verify=>false)`, `inverse(f,Verify=>Probabilistic)`, or `inverse(f,Verify=>true)` would make everything clear to the reader.  Unfortunately, `inverse` doesn't accept optional arguments, but you can simulate it by installing a method for `inverse(RationalMap,Option)`, anticipating the ultimate day when we take the time to convert `inverse` to a function that does accept options.

Wouldn't that be better?

Reply:
`inverseMap(...,MathMode=>false)` performs no test at all, and  `inverse` is basically the same of `inverseMap(...,MathMode=>true)`, which performs a full test. So I see `inverse'` as a middle ground approach between MathMode=>true and MathMode=>false.

Reply:
That's even more confusing.  Wouldn't it be a good idea to put the functionality of `inverseMap` into a method for `inverse`?

Reply:
> In any case, this sort of thing is exactly why we have optional named arguments to some method functions. Writing something like `inverse(f,Verify=>false)`, `inverse(f,Verify=>Probabilistic)`, or `inverse(f,Verify=>true)` would make everything clear to the reader. Unfortunately, `inverse` doesn't accept optional arguments, but you can simulate it by installing a method for `inverse(RationalMap,Option)`, anticipating the ultimate day when we take the time to convert `inverse` to a function that does accept options.
> 
> Wouldn't that be better?

I'll try to do as you say, thanks.

Reply:
> That's even more confusing. Wouldn't it be a good idea to put the functionality of `inverseMap` into a method for `inverse`?

I agree with you. The same type of problem occurs with other functions, such as degreeOfRationalMap which could be degree(RationalMap, MathMode => false/true). But in the beginning I had implemented all the methods only for the RingMaps class (interpreted as rational maps) and methods such as degree(RingMap) and inverse(RingMap) already existed. Now, considering that the package is several years old, I would prefer to continue accepting ring maps as input.

Reply:
Okay.   But now what about the name `MathMode`?  That's not descriptive, because all computations in M2 are done with "math".  What about Verify instead?

Reply:
> Okay. But now what about the name `MathMode`? That's not descriptive, because all computations in M2 are done with "math". What about Verify instead?

Yes, but for this version it would take me too much effort to change that name. However, the name was chosen to be the same on all methods of the package. It can mean checking the output (as in inverseMap) but also performing the calculation using a non-probabilistic algorithm (as in projectiveDegrees).

Reply:
> In any case, this sort of thing is exactly why we have optional named arguments to some method functions. Writing something like `inverse(f,Verify=>false)`, `inverse(f,Verify=>Probabilistic)`, or `inverse(f,Verify=>true)` would make everything clear to the reader. Unfortunately, `inverse` doesn't accept optional arguments, but you can simulate it by installing a method for `inverse(RationalMap,Option)`, anticipating the ultimate day when we take the time to convert `inverse` to a function that does accept options.
> 
> Wouldn't that be better?

I did this and it looks nicer indeed.

Reply:
> Okay. But now what about the name `MathMode`? That's not descriptive, because all computations in M2 are done with "math". What about Verify instead?

A better name than `MathMode` might be for example be `Certify`. But I would like not to deviate too much from the certified version of the package. However I really appreciate the comments.